dropwizard-logging/src/main/java/io/dropwizard/logging/FileAppenderFactory.java

```diff
@@ -220,16 +220,19 @@ public boolean isValidFileConfiguration() {
 
             if (maxFileSize != null && !archivedLogFilenamePattern.contains("%d")) {
                 final FixedWindowRollingPolicy rollingPolicy = new FixedWindowRollingPolicy();
-                final SizeBasedTriggeringPolicy<E> triggeringPolicy = new SizeBasedTriggeringPolicy<>();
-                triggeringPolicy.setMaxFileSize(String.valueOf(maxFileSize.toBytes()));
-                triggeringPolicy.setContext(context);
                 rollingPolicy.setContext(context);
                 rollingPolicy.setMaxIndex(getArchivedFileCount());
                 rollingPolicy.setFileNamePattern(getArchivedLogFilenamePattern());
-                appender.setRollingPolicy(rollingPolicy);
-                appender.setTriggeringPolicy(triggeringPolicy);
                 rollingPolicy.setParent(appender);
                 rollingPolicy.start();
+                appender.setRollingPolicy(rollingPolicy);
+                
+                final SizeBasedTriggeringPolicy<E> triggeringPolicy = new SizeBasedTriggeringPolicy<>();
+                triggeringPolicy.setMaxFileSize(String.valueOf(maxFileSize.toBytes()));
+                triggeringPolicy.setContext(context);
+                triggeringPolicy.start();
```

Comment:
Posted under wrong thread, reposting in the correct context.

I think there are some other issues[1] with the implementation in the else block. It seems like the `%d-%i` configuration should've never been supported. I believe there will need more intrusive changes in order to get a more robust rolling/triggering policy in those workflows.

[1] http://jira.qos.ch/browse/LOGBACK-1127?focusedCommentId=17110&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-17110
[2] http://jira.qos.ch/browse/LOGBACK-1143